From b0b69e739495cfb32da0fcc14cd2f74db374f512 Mon Sep 17 00:00:00 2001
From: tpgxyz <tpgxyz@gmail.com>
Date: Fri, 26 Feb 2021 20:33:02 +0100
Subject: [PATCH 0/1] try to retry DNS catch

---
 dns_checker.c | 21 ++++++++++++++++-----
 dns_checker.h |  2 ++
 2 files changed, 18 insertions(+), 5 deletions(-)

diff --git a/dns_checker.c b/dns_checker.c
index 193850b..87bbf1e 100644
--- a/dns_checker.c
+++ b/dns_checker.c
@@ -9,11 +9,22 @@ static struct addrinfo hints, *infoptr;
 int check_dns() {
 	hints.ai_family = AF_UNSPEC;
 	hints.ai_socktype = SOCK_STREAM;
-	hints.ai_flags = AI_PASSIVE; 
-	int result = getaddrinfo("github.com", NULL, &hints, &infoptr);
-	if (result) {
-		log_printf(LOG_FATAL, "getaddrinfo: %s\n", gai_strerror(result));
-		return -1;
+	hints.ai_flags = AI_PASSIVE;
+	for (int try = 0; try < DNS_RETRIES; try++) {
+	    log_printf(LOG_INFO, "Try #%d: Resolving DNS.\n", try + 1);
+	
+	    int result = getaddrinfo("github.com", NULL, &hints, &infoptr);
+	    if (result) {
+		    log_printf(LOG_INFO, "getaddrinfo: %s\n", gai_strerror(result));
+		    if ( try < DNS_RETRIES ) {
+			log_printf(LOG_ERROR, "Sleeping for 5 seconds.\n");
+			sleep(5);
+		    } else {
+			    return -1;
+			}
+	    } else {
+		    try = DNS_RETRIES;
+		}
 	}
 
 	log_printf(LOG_DEBUG, "github.com was resolved to:\n");
diff --git a/dns_checker.h b/dns_checker.h
index 3d792aa..98b14d9 100644
--- a/dns_checker.h
+++ b/dns_checker.h
@@ -1,6 +1,8 @@
 #ifndef _DNS_CHECKER_H
 #define _DNS_CHECKER_H
 
+#define DNS_RETRIES 10
+
 #include "log_levels.h"
 
 int check_dns();
-- 
2.30.1

